superset/daos/database.py

```diff
@@ -166,6 +170,44 @@ def get_ssh_tunnel(cls, database_id: int) -> SSHTunnel | None:
 
         return ssh_tunnel
 
+    @classmethod
+    def is_odps_partitioned_table(
```

Comment:
all of the database-engine-specific code belongs in this package -> https://github.com/apache/superset/tree/master/superset/db_engine_specs

There should be hooks / methods related to partition-management in there. You'll probably need a proper DBAPI driver and sqlalchemy dialect. Guessing pyodps package provides that (?)